### ISSUE TITLE
Issue with scrollbars and horizontal scrolling

### ISSUE THREAD

**Issue by MuhammedZakir** [fjvallarino/monomer#136]
1. Horizontal scrolling using `shift + mouse wheel` is not supported?
2. Scrolling *when hovering* over a scrollbar should scroll that one. Currently, vertical scrollbar is scrolled even if horizontal one is hovered.

Edit: FYI, I only used this on Linux.

---

Thanks for this library. It's really nice. Especially how easy using this is; not something usually focused on, so really appreciate this. :-)

**Comment by fjvallarino** [comment]
Hi @MuhammedZakir!

I had not noticed `shift + mouse wheel` was supported cross-platform; I added it now. You can test it by modifying your `stack.yaml` file to point to:

```yaml
- git: https://github.com/fjvallarino/monomer.git
  commit: ccdc3bdf2bf372dd391cddf2e0403f2f03fb7506
```

This commit hash will no longer be valid once the PR is merged to main; I'll update the issue then.

Regarding the second point, scrolling based on the hovered scrollbar, I have not found it to be the standard behavior. I tried it on macOS, Windows, and Ubuntu (GNOME), and in all cases using the vertical wheel while hovering the horizontal scrollbar resulted in vertical scrolling. Curious: which desktop manager are you using? Could it be that it is something specific to your setup?

---
Thanks for the kind words! One of the main objectives of the library is to be easy to use, so it's nice to read that you think it accomplishes that goal.

**Comment by MuhammedZakir** [comment]
> Hi @MuhammedZakir!

Hello! :-)
 
> I had not noticed `shift + mouse wheel` was supported cross-platform; I added it now. You can test it by modifying your `stack.yaml` file to point to:
> 
> ```yaml
> - git: https://github.com/fjvallarino/monomer.git
>   commit: ccdc3bdf2bf372dd391cddf2e0403f2f03fb7506
> ```
> 
> This commit hash will no longer be valid once the PR is merged to main; I'll update the issue then.

It's working. Thanks for the quick fix.

> Regarding the second point, scrolling based on the hovered scrollbar, I have not found it to be the standard behavior. I tried it on macOS, Windows, and Ubuntu (GNOME), and in all cases using the vertical wheel while hovering the horizontal scrollbar resulted in vertical scrolling. Curious: which desktop manager are you using? Could it be that it is something specific to your setup?

I hadn't used Windows and macOS for sometime now. But I remember doing that. I'll report back after testing.

Regarding Linux: I am using Manjaro XFCE. I have tested this in a browser (Luakit), GParted, Inkscape and XFCE Task Manager. I think I have seen the same behavior in KDE. But like the other OSes, I haven't used it for a while, so I don't know if I am remebering it wrongly.

> Thanks for the kind words! One of the main objectives of the library is to be easy to use, so it's nice to read that you think it accomplishes that goal.

You're welcome! :-)

---

Out of topic: in the Monomer lens utils, it would be nice if you could add a one-liner description to each function. For example, when I read `Monomer.Graphics.Lens`, I was confused when I saw `a` function. I guessed it is to get the alpha value after looking at other functions in that module. The same happened when I read the doc for the widget-lens-util module. I could open a PR, but that will take a long time as I will need to familiarize with all these first.

---

In your examples, why did you use `makeLenses '<data constructor>` instead of `makeLenses ''<type>`? FYI, I am a beginner Haskeller.

**Comment by fjvallarino** [comment]
One other issue regarding the hovered scrollbar is that SDL sends the same event for both wheel and trackpad movement (there is no way for discerning which of them happened). In my experience, trackpads always follow the direction of movement regardless of what is hovered. Considering this, implementing this behavior would be confusing for trackpad users.

---
Regarding the lens-related question, there is not really a good reason; I never use multiple constructors for record types, so in this case it is the same.

**Comment by fjvallarino** [comment]
The PR has been merged to main, so the previous commit hash is no longer valid. The new hash, which will not be removed, is this:

```yaml
- git: https://github.com/fjvallarino/monomer.git
  commit: 80fa65831e19ec7a862cc905e404b208dfebff8d
```

**Comment by MuhammedZakir** [comment]
> In my experience, trackpads always follow the direction of movement regardless of what is hovered. Considering this, implementing this behavior would be confusing for trackpad users.

FYI, when i tested this in Manjaro XFCE, it works just like mouse. That is, when I hover a horizontal scrollbar and do vertical swipe on trackpad, it scrolls the horizontal scrollbar.

Honestly, this is wierd we're getting totally different behavior (though not unexpected or surprising). I will test it on macOS and Windows soon. Can you mention some apps you used to test this?

**Comment by fjvallarino** [comment]
I tried with the Preview app, Pixelmator, VS Code, and Chrome on macOS. On Windows with VS Code and Chrome; same on Ubuntu.

Testing horizontal scrolling with a visible scrollbar in a browser is tricky since most content nowadays is responsive. I tried, for example, with this Gist since it has a forced container width: https://github.com/phaazon/stack-haddock-upload/blob/master/stack-haddock-upload
